streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsMetadataState.java

```diff
@@ -324,6 +345,45 @@ private void rebuildMetadata(final Map<HostInfo, Set<TopicPartition>> activePart
             return;
         }
 
+        allMetadata = topologyMetadata.hasNamedTopologies() ?
+            rebuildMetadataForNamedTopologies(activePartitionHostMap, standbyPartitionHostMap) :
+            rebuildMetadataForSingleTopology(activePartitionHostMap, standbyPartitionHostMap);
+    }
+
+    private List<StreamsMetadata> rebuildMetadataForNamedTopologies(final Map<HostInfo, Set<TopicPartition>> activePartitionHostMap,
+                                                                    final Map<HostInfo, Set<TopicPartition>> standbyPartitionHostMap) {
+        final List<StreamsMetadata> rebuiltMetadata = new ArrayList<>();
+        Stream.concat(activePartitionHostMap.keySet().stream(), standbyPartitionHostMap.keySet().stream())
+            .distinct()
+            .forEach(hostInfo -> {
+                final Set<TopicPartition> activePartitionsOnHost = new HashSet<>();
+                final Set<TopicPartition> standbyPartitionsOnHost = new HashSet<>();
+
+                final Map<String, Set<String>> storeNamesByTopology = new HashMap<>();
+                final Map<String, Set<String>> standbyStoreNamesByTopology = new HashMap<>();
+
+                topologyMetadata.namedTopologiesView().forEach(topologyName -> {
+                    // WIP 
+                });
+
+                final NamedTopologyStreamsMetadataImpl metadata = new NamedTopologyStreamsMetadataImpl(
+                    hostInfo,
+                    storeNamesByTopology,
```

Comment:
Since this PR is not completed yet (btw could you rename it to indicate it's WIP?), I'm not sure what's the motivation from changing it to a mapping from topology to names, but I feel we may not need to maintain two mappings for active and standby since we are always trying to check if store name is in either of them, is that right?

Reply:
See the latest changes -- lmk if you still have a question here